### PR TITLE
Upgrade machine-emulator base image

### PR DESCRIPTION
**Pull request by vfusco**
- Upgrade machine-emulator base image (Debian)
- Upgrade grpc-interfaces to v0.14.0
- Build for amd64 and arm64 using depot.dev

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,59 +31,51 @@ SERVER_MANAGER_TO_BIN= server-manager
...
 all: source-default
 
 env:
 	@echo $(LIBRARY_PATH)
-	@echo "export PATH='$(SRCDIR):$(BUILDDIR)/bin:/opt/cartesi/bin:${PATH}'"
-	@echo "export LUA_CPATH='./?.so;$(SRCDIR)/?.so;/opt/cartesi/lib/lua/5.3/?.so;$$(lua5.3 -e 'print(package.cpath)')'"
-	@echo "export LUA_PATH='./?.lua;$(SRCDIR)/?.lua;/opt/cartesi/share/lua/5.3/?.lua;$$(lua5.3 -e 'print(package.path)')'"
+	@echo "export PATH='$(SRCDIR):/opt/cartesi/bin:${PATH}'"
+	@echo "export LUA_CPATH_5_4='./?.so;$(SRCDIR)/?.so;/opt/cartesi/lib/lua/5.4/?.so;$$(lua5.4 -e 'print(package.cpath)')'"
+	@echo "export LUA_PATH_5_4='./?.lua;$(SRCDIR)/?.lua;/opt/cartesi/share/lua/5.4/?.lua;$$(lua5.4 -e 'print(package.path)')'"
```
Comment (edubart) [inline Makefile]: Here should do the same way latest `machine-emulator` is doing on its `env` target, using the `${LUA_PATH_5_4:-;}` trick.
Also for some systems cartesi machine will not be installed in `/opt/cartesi`, so maybe change `/opt/cartesi/` to `$(PREFIX)`

File: src/Makefile [hunk 2]
```diff
@@ -58,13 +60,15 @@ CRYPTOPP_INC_Darwin:=-I/opt/local/include
...
 BOOST_FILESYSTEM_LIB_Darwin:=$(BOOST_LIB_DIR_Darwin) -lboost_system-mt -lboost_filesystem-mt
 BOOST_LOG_LIB_Darwin:=$(BOOST_LIB_DIR_Darwin) -lboost_log-mt -lboost_log_setup-mt -lboost_thread-mt
 BOOST_PROCESS_LIB_Darwin:=-lpthread
-GRPC_INC_Darwin=-I$(BUILDDIR)/include
-GRPC_LIB_Darwin=-L$(BUILDDIR)/lib -lgrpc++ -lgrpc -lgpr -lprotobuf -lpthread -labsl_synchronization
-PROTOBUF_LIB_Darwin:=-L$(BUILDDIR)/lib -lprotobuf -lpthread
+GRPC_INC_Darwin=-I/opt/local/include
+GRPC_LIB_Darwin=-L/opt/local/lib -lgrpc++ -lgrpc -lgpr -lprotobuf -lpthread -labsl_synchronization
+PROTOBUF_LIB_Darwin:=-L/opt/local/lib -lprotobuf -lpthread
 CARTESI_EXECUTABLE_LDFLAGS_Darwin=-Wl,-rpath,$(BUILDDIR)/lib -Wl,-rpath,$(CURDIR)
```
Comment (edubart) [inline src/Makefile:73]: I need the flag ` -Wl,--copy-dt-needed-entries` to compile here.

File: Makefile (hunk 1 above)
Comment (edubart) [reply Makefile]: I pushed a fixup with the change.

File: src/Makefile (hunk 2 above)
Comment (edubart) [reply src/Makefile:73]: I pushed a fixup with the change.

File: Makefile [hunk 3]
```diff
@@ -31,59 +31,51 @@ SERVER_MANAGER_TO_BIN= server-manager
...
 
 doc:
 	cd doc && doxygen Doxyfile
 
 help:
 	@echo 'Common targets:'
-	@echo '* all                        - build the src/ code. To build from a clean clone, run: make submodules dep all'
+	@echo '* all                        - build the src/ code. To build from a clean clone, run: make submodules all'
 	@echo '  submodules                 - initialize and update submodules'
-	@echo '  dep                        - build dependencies'
```
Comment (edubart) [inline Makefile:86]: The readme need updates about the new dependencies vrsions (gRPC and Lua). Also change Ubuntu instructions to Debian.

**Review by edubart — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -15,241 +21,454 @@ jobs:
...
+          docker run --rm -p 5001:5001 -h server-manager-container --name server-manager-container --network mynetwork -v server-manager-root:/tmp/server-manager-root ${{ github.repository_owner }}/server-manager:devel &
           timeout 20 bash -c 'while ! nc -q0 127.0.0.1 5001 < /dev/null > /dev/null 2>&1; do sleep 1; done'
-          docker run --rm -u cartesi:cartesi -h test-container --name test-container --network mynetwork -v /opt/cartesi/share/images:/opt/cartesi/share/images -v server-manager-root:/tmp/server-manager-root ${{ secrets.DOCKER_ORGANIZATION }}/server-manager:installer make run-test-server-manager MANAGER_ADDRESS=server-manager-container:5001
+          docker run --rm -u cartesi:cartesi -h test-container --name test-container --network mynetwork -v /tmp/cartesi-machine-images:${{ env.CARTESI_IMAGES_PATH }} -v server-manager-root:/tmp/server-manager-root ${{ github.repository_owner }}/server-manager:installer make run-test-server-manager MANAGER_ADDRESS=server-manager-container:5001
           docker kill server-manager-container
 
-  test-echo-dapp:
-    name: Test with rollup-init echo-dapp
+  test-ioctl-echo-loop-arm64:
```
Comment (mpolitzer) [inline .github/workflows/build.yml]: Consider using matrix setup for this and `test-ioctl-echo-loop-amd64` jobs.
Same comment applies to other `-amd64`, `-arm64` pairs.

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -15,241 +21,454 @@ jobs:
...
+          cd /tmp/cartesi-machine-images/ && ln -s linux-${{ env.LINUX_VERSION }}.bin linux.bin
+          cd /tmp/cartesi-machine-images/ && ln -s rom-${{ env.ROM_VERSION }}.bin rom.bin
+          cd /tmp/cartesi-machine-images/ && ln -s rootfs-${{ env.ROOTFS_VERSION }}.ext2 rootfs.ext2
+
+      - name: Run test suite inside the docker image (Linux/ARM64)
+        run: |
+          docker volume create --name server-manager-root
+          docker network create mynetwork
+          docker run --platform linux/arm64 --rm -v server-manager-root:/tmp/server-manager-root ${{ github.repository_owner }}/server-manager:installer chown -R cartesi:cartesi /tmp/server-manager-root
+          docker run --platform linux/arm64 --rm -u cartesi:cartesi -v /tmp/cartesi-machine-images:${{ env.CARTESI_IMAGES_PATH }} -v server-manager-root:/tmp/server-manager-root ${{ github.repository_owner }}/server-manager:installer make create-machines
```
Comment (mpolitzer) [inline .github/workflows/build.yml]: Consider having a variable for this path (`/tmp/cartesi-machine-images`).
Something like `HOST_IMAGES_PATH`.

File: src/Makefile [hunk 6]
```diff
@@ -58,13 +60,15 @@ CRYPTOPP_INC_Darwin:=-I/opt/local/include
 endif
 endif
 
+PROTOC_Darwin:=/opt/local/bin/protoc
```
Comment (mpolitzer) [inline src/Makefile]: No need to hardcode the PATH if we can be relative to `grpc_cpp_plugin`.
I would go with something like this.

```
GRPC_CPP_PLUGIN_Darwin:=$(shell which grpc_cpp_plugin)
GRPC_INC_Darwin:=-I$(dir $(GRPC_CPP_PLUGIN_Darwin))../include
GRPC_LIB_Darwin:=-L$(dir $(GRPC_CPP_PLUGIN_Darwin))../lib -lgrpc++ -lgrpc -lgpr -lprotobuf -lpthread -labsl_synchronization
PROTOBUF_LIB_Darwin:=-L$(dir $(GRPC_CPP_PLUGIN_Darwin))../lib -lprotobuf -lpthread
```

File: src/Makefile [hunk 7]
```diff
@@ -73,6 +77,8 @@ SOLDFLAGS_Linux:=-shared -fPIC -pthread
 CC_Linux=gcc
 CXX_Linux=g++
 INCS_Linux=
+PROTOC_Linux:=/usr/bin/protoc
```
Comment (mpolitzer) [inline src/Makefile]: Same trick as above here and on the next line.

File: src/server-manager.cpp [hunk 8]
```diff
@@ -100,7 +100,7 @@ constexpr const uint64_t ROLLUP_ADVANCE_STATE = 0;
 constexpr const uint64_t ROLLUP_INSPECT_STATE = 1;
 
 static constexpr uint32_t manager_version_major = 0;
-static constexpr uint32_t manager_version_minor = 7;
+static constexpr uint32_t manager_version_minor = 8;
```
Comment (mpolitzer) [inline src/server-manager.cpp:103]: Should we open an issue to handle this in the Makefile, the same as we do in `machine-emulator`?

File: src/test-server-manager.cpp [hunk 9]
```diff
@@ -569,12 +565,12 @@ void assert_bool(bool value, const std::string &msg, const std::string &file, in
 #define ASSERT_STATUS_CODE(s, f, v) assert_status_code(s, f, v, __FILE__, __LINE__)
 
 static void test_get_version(const std::function<void(const std::string &title, test_function f)> &test) {
-    test("The server-manager server version should be 0.7.x", [](ServerManagerClient &manager) {
+    test("The server-manager server version should be 0.8.x", [](ServerManagerClient &manager) {
```
Comment (mpolitzer) [inline src/test-server-manager.cpp:568]: These versions should be automatically fetched from the build. Can we have an issue for that as well?

File: src/server-manager.cpp (hunk 8 above)
Comment (vfusco) [reply src/server-manager.cpp:103]: Yes please

File: src/test-server-manager.cpp (hunk 9 above)
Comment (vfusco) [reply src/test-server-manager.cpp:568]: Yes please

**Review by diegonehab — DISMISSED** [review]

File: src/Makefile (hunk 6 above)
Comment (mpolitzer) [reply src/Makefile]: sent you a patch

File: src/server-manager.cpp (hunk 8 above)
Comment (mpolitzer) [reply src/server-manager.cpp:103]: Created an issue for it: https://github.com/cartesi/server-manager/issues/10

**Review by edubart — DISMISSED** [review]

**Review by mpolitzer — DISMISSED** [review]

**Review by mpolitzer — APPROVED** [review]

**Review by edubart — APPROVED** [review]

**Pull request merged**